core-services/prow/03_deployment/boskos_cleaner.yaml

```diff
@@ -0,0 +1,41 @@
+# Boskos cleaner is needed to allow resources created out of dynamic resource lifecycle to
+# be deleted.
+# It is responsible for checking if resources that were created based on
+# dynamic resource lifecycles have child resources defined in their userdata and if yes,
+# mark those as to be deleted as well.
+# Afterwards the resources are put into tombstone state which makes boskos delete them
+# the next time it syncs its config.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: boskos-cleaner
+  labels:
+    app: prow
+    component: boskos-cleaner
+  namespace: ci
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      component: boskos-cleaner
+  template:
+    metadata:
+      labels:
+        app: prow
+        component: boskos-cleaner
+    spec:
+      priorityClassName: prow-critical
+      serviceAccountName: boskos
+      terminationGracePeriodSeconds: 300
+      containers:
+      - name: boskos-cleaner
+        image: gcr.io/k8s-prow/boskos/cleaner:v20200218-a445c54db
+        args:
+        - --cleaner-count=1
```

Comment:
It does an acquire once per worker per hardcoded 15 seconds interval. In order to give poor lock contained boskos a bit of slack, we use only one worker.